### DISCUSSION TITLE
IDE for project

### DISCUSSION THREAD

**Discussion by spritz-dot** [nextgenhealthcare/connect#4517] [Q&A]
Hi all. Does anyone use any IDE (Intellij is my case) to run JavaScript code using the MIRTH environment? I found online two guides (pretty old) but I cannot figure out how to make it works. Thanks for your answer.

**Comment by tom08zehn** [comment]
What exactly are you trying to achieve?

To test your code you always need to run it in Mirth itself. I tried a lot of different things and really was close but finally gave up because it's impossible to "simulate" a _full_ Mirth environment... then you could rather use Eclipse and run Mirth's source code in there.

If you just need a better IDE than Mirth's JavaScript windows then you could also use external JS files, e.g. for the transformers, and edit the code in your preferred editor or IDE. If your editor supports to run a batch file via shortcut then you could run a script that uses the Mirth API in order to redeploy your channel and reprocess a test message after you modified your JS code (example here: https://github.com/nextgenhealthcare/connect/discussions/4528)

Once I finished coding I copy the JS code back into the channel because it's easier to maintain and deploy an all-in-one channel-file in comparison to a channel file plus multiple external JS files.
